packages/backend/src/rootContext.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  CacheManager,
+  cacheManagerDep,
+  DatabaseManager,
+  databaseManagerDep,
+  pluginEndpointDiscoveryDep,
+  ServerTokenManager,
+  SingleHostDiscovery,
+  tokenManagerDep,
+  urlReaderDep,
+  UrlReaders,
+} from '@backstage/backend-common';
+import { createDependencyConfig } from '@backstage/app-context-common';
+import { Config, configDep } from '@backstage/config';
+import { Logger } from 'winston';
+import { permissionAuthorizerDep } from '@backstage/plugin-permission-common';
+import { ServerPermissionClient } from '@backstage/plugin-permission-node';
+import { TaskScheduler, taskSchedulerDep } from '@backstage/backend-tasks';
+
+export const rootDependencies = (config: Config, root: Logger) => {
```

Comment:
Am I understanding this function correctly that, if I have my own custom services that I wanted to add to the application context and have accessible via `container.get<Foo>(fooDep)`, all I would need to do is add an item to this array with

```ts
createDependencyConfig({
  id: fooDep,
  dependencies: {
    config: configDep,
    bar: barDep
  },
  factory: (config, bar) => new FooDep(config, bar)
})
```

and, assuming `Bar` was also added, I'd be good-to-go, right?

Reply:
Correct yeah.
The dependencies are passed in to the factory in the same shape (`({})`, within an object, since we can piggyback on JS object construction possibilities) as long as they can be found from the container.